src/git.rs

```diff
@@ -50,7 +50,13 @@ pub fn generate_commit_msg(survey: SurveyResults) -> String {
         None => with_long_msg,
     };
     match survey.affected_open_issues {
-        Some(issue_list) => format!("{}\n\n{}", with_breaking_change, issue_list),
+        Some(issue_list) => {
+            let prefix = match &survey.commit_type[..] {
+                "fix" => "Fixes: ",
+                _ => "Referenced issues: ",
```

Comment:
The [specification](https://www.conventionalcommits.org/en/v1.0.0/#specification) says that footer's word token is separated by dashes, meaning that it should be `Referenced issues` or maybe even `Issues` instead.

Additionally, the separator between a footer's word token and its token is either `:<space>` or `<space> #`. Meaning that if a user enters `#123`, this is actually wrongly formatted as the line would be `Fixes: #123` but has to be `Fixes #123` in this case. The alternative would be to leave the `#` out, then this method would be correct.

This means that an additional check for the separator has to be done. If a hashtag is present, leave the `:` out, if not, append it.

__Edit:__ I know I know, it's annoying, but I want to stay close to the specification to allow other automated tools to pick up on the commits that `git-cz` generates :)

Reply:
It's not annoying and you're definitely right to want to stick to the specification, I just assumed that it was correct since github allows `Fixes: #<id>`

So `fix <id>` (which seems to be recommended) and `Referenced-issues <id>` are both correct?

Also it seems like `feat <id>` is a thing for new features

Reply:
Ye, you have to use a colon unless the next work starts with a hashtag, aka is an issue reference number:
- `Fixes: 123` is theoretically the same as `Fixes #123`. However, the second is better since it conveys the intention behind it, an issue reference.
- `Referenced-issue: 123` is the same as `Referenced-issue #123`. Same argument as above here.

I mean it's all just convention. If a user enters `#some string`, it's his problem to take care off :) But I would like to at least check for the hashtag and if non is present, use a colon.

On that note, what happens if someone enters `This is the prompt we ask> ::space entered by user::#123`. If I remember the way I implemented it correctly, this would result in `Fixes:  #123`, as no whitespace is trimmed from my side. But maybe `dialoguer` trims it already... 🤔

Reply:
The parsing of the footer is described here:
> 8. One or more footers MAY be provided one blank line after the body. Each footer MUST consist of a word token, followed by either a :<space> or <space># separator, followed by a string value (this is inspired by the git trailer convention).
10. A footer’s value MAY contain spaces and newlines, and parsing MUST terminate when the next valid footer token/separator pair is observed.

Given point no. 10, I understand it that way that if both are specified, `Fixes: #123`, this wouldn't work. But maybe I am wrong here... Would be cool if you could read over footer-related points and see if I am correct here. Before you put any more work into it and it turns out to not be right :D

Reply:
Fixes: #123 would then basically me one valid footer word match (`Fixes: `) but I am not sure if this would mean that it would detect the second part as another footer word (` #123`).

Reply:
I'll see to modify the PR so that it has the following behavior:
- If the user inputs non-conventional issues, then it's his problem
- Now uses the following tags: `fix`, `feat` and defaults to `Referenced-issues`
- Entered issue references are space-separated (indicated in the prompt: `? Add issue references (space-separated):`)
- If an issue starts with `#`, then use ` `
- If an issue doesn't start with `#`, then use `: `

Should a list of issues result in one line per issue? The specification says this, which isn't very precise:
> A footer’s value MAY contain spaces and newlines, and parsing MUST terminate when the next valid footer token/separator pair is observed.

Reply:
Ye that's the part I do not understand 100% to be honest. In an ideal world, there would be one commit per issue/pr only, so the problem wouldn't arise. But I couldn't find any resources on how to handle it for multiple. That being said, I would put this into the same drawer as earlier on. It's up to the user to enter it right. 

I wouldn't enter two issues at once in almost all cases, but if he wants to, it's up to him! :) I think in that sense it's better to use the value as it is supplied by the user.

Reply:
>     * Now uses the following tags: `fix`, `feat` and defaults to `Referenced-issues`

I would suggest to use `Fixes` for fix and `Closes` for feat.

Reply:
Oh right, I mixed it up with the commit `type`s. Alright